pipeline.go

```diff
@@ -44,6 +44,12 @@ type Source interface {
 	Record() (interface{}, error)
 }
 
+// SourcePeeker is an interface for peeking ahead at the next record
+// to be returned by Source.Record().
+type SourcePeeker interface {
```

Comment:
I would have gone with `Peeker` I think... though I can see an argument either way.

Reply:
makes sense to me. i'll change it.